lib/node_modules/@stdlib/blas/base/drot/test/test.drot.js

```diff
@@ -0,0 +1,325 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var Float64Array = require( '@stdlib/array/float64' );
+var dcopy = require( '@stdlib/blas/base/dcopy' );
+var EPS = require( '@stdlib/constants/float64/eps' );
+var abs = require( '@stdlib/math/base/special/abs' );
+var drot = require( './../lib/drot.js' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof drot, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function has an arity of 7', function test( t ) {
+	t.strictEqual( drot.length, 7, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the function applies a plane rotation', function test( t ) {
+	var xexpected;
+	var yexpected;
+	var ixvalues;
+	var iyvalues;
+	var nvalues;
+	var xvalues;
+	var yvalues;
+	var delta;
+	var tol;
+	var out;
+	var ex;
+	var ey;
+	var ix;
+	var iy;
+	var c;
+	var i;
+	var j;
+	var k;
+	var x;
+	var y;
+
+	xexpected = [
+		[ 0.6, 0.10, -0.5, 0.8, 0.9, -0.3, -0.4 ],
+		[ 0.78, 0.10, -0.5, 0.8, 0.9, -0.3, -0.4 ],
+		[ 0.78, -0.46, -0.5, 0.8, 0.9, -0.3, -0.4 ],
+		[ 0.78, -0.46, -0.22, 1.06, 0.9, -0.3, -0.4 ],
+		[ 0.6, 0.10, -0.5, 0.8, 0.9, -0.3, -0.4 ],
+		[ 0.78, 0.10, -0.5, 0.8, 0.9, -0.3, -0.4 ],
+		[ 0.66, 0.1, -0.1, 0.8, 0.9, -0.3, -0.4 ],
+		[ 0.96, 0.1, -0.76, 0.8, 0.9, -0.3, -0.020000000000000073],
+		[ 0.6, 0.10, -0.5, 0.8, 0.9, -0.3, -0.4 ],
+		[ 0.78, 0.10, -0.5, 0.8, 0.9, -0.3, -0.4 ],
+		[ -0.06000000000000005, 0.1, -0.1, 0.8, 0.9, -0.3, -0.4 ],
+		[ 0.9, 0.1, -0.22, 0.8, 0.18, -0.3, -0.020000000000000073],
+		[ 0.6, 0.10, -0.5, 0.8, 0.9, -0.3, -0.4 ],
+		[ 0.78, 0.10, -0.5, 0.8, 0.9, -0.3, -0.4 ],
+		[ 0.78, 0.26, -0.5, 0.8, 0.9, -0.3, -0.4 ],
+		[ 0.78, 0.26, -0.76, 1.12, 0.9, -0.3, -0.4 ]
+	];
+	yexpected = [
+		[ 0.5, -0.9, 0.3, 0.7, -0.6, 0.2, 0.8 ],
+		[ 0.040000000000000036, -0.9, 0.3, 0.7, -0.6, 0.2, 0.8 ],
+		[ 0.040000000000000036, -0.78, 0.3, 0.7, -0.6, 0.2, 0.8 ],
+		[ 0.040000000000000036, -0.78, 0.54, 0.07999999999999996, -0.6, 0.2, 0.8 ], // eslint-disable-line max-len
+		[ 0.5, -0.9, 0.3, 0.7, -0.6, 0.2, 0.8 ],
+		[ 0.040000000000000036, -0.9, 0.3, 0.7, -0.6, 0.2, 0.8 ],
+		[ 0.7, -0.9, -0.12, 0.7, -0.6, 0.2, 0.8 ],
+		[ 0.64, -0.9, -0.3, 0.7, -0.18, 0.2, 0.28000000000000014 ],
+		[ 0.5, -0.9, 0.3, 0.7, -0.6, 0.2, 0.8 ],
+		[ 0.040000000000000036, -0.9, 0.3, 0.7, -0.6, 0.2, 0.8 ],
+		[ 0.7, -1.08, 0.3, 0.7, -0.6, 0.2, 0.8 ],
+		[ 0.64, -1.26, 0.54, 0.2, -0.6, 0.2, 0.8 ],
+		[ 0.5, -0.9, 0.3, 0.7, -0.6, 0.2, 0.8 ],
+		[ 0.040000000000000036, -0.9, 0.3, 0.7, -0.6, 0.2, 0.8 ],
+		[ 0.040000000000000036, -0.9, 0.18, 0.7, -0.6, 0.2, 0.8 ],
+		[ 0.040000000000000036, -0.9, 0.18, 0.7, -0.18, 0.2, 0.16000000000000014 ] // eslint-disable-line max-len
+
+	];
+	nvalues = [ 0, 1, 2, 4 ];
+	ixvalues = [ 1, 2, -2, -1 ];
+	iyvalues = [ 1, -2, 1, -2 ];
```

Comment:
I would much prefer we split these out into separate `tape() {...}` test blocks. Having nested arrays like this makes debugging tests more difficult.